utils/utils.go

```diff
@@ -809,6 +809,17 @@ func CustomizeServiceAccount(sa *corev1.ServiceAccount, ba common.BaseComponent,
 			}
 		}
 	}
+
+	mount := true
+	basa := ba.GetServiceAccount()
+	if basa != nil {
+		if basa.GetMountSAToken() != nil && !*basa.GetMountSAToken() {
+			// not nil and set to false
+			mount = false
+		}
+	}
+	sa.AutomountServiceAccountToken = &mount
+
```

Comment:
We should set `automountServiceAccountToken: false` on Pod as well. This way, user can disable mount even for the non-default SA.

Reply:
@leochr 
Can I clarify this?
Is the suggestion that we will have a single config item (.spec.ServiceAccount.MountToken), and if that is set to false, then we will set both 
 * AutomountServiceAccountToken=false on the service account we create
 * automountServiceAccountToken=false on the deployment pod spec
 
Alternatively we could just set this _only_ on the pod spec. That should cover both the case of using our service account or a user specified one, and would make the code simpler.

Reply:
@idlewis It's fine to set it only on the pod spec (as it takes precedence over SA)